Remotion/Core/Reflection/MethodInfoAdapter.cs

```diff
@@ -57,25 +57,25 @@ public static MethodInfoAdapter Create (MethodInfo methodInfo)
     }
 
     private readonly MethodInfo _methodInfo;
-    private readonly Lazy<ITypeInformation> _cachedDeclaringType;
+    private readonly Lazy<ITypeInformation?> _cachedDeclaringType;
     private readonly Lazy<ITypeInformation> _cachedOriginalDeclaringType;
 
-    private readonly Lazy<IPropertyInformation> _declaringProperty;
+    private readonly Lazy<IPropertyInformation?> _declaringProperty;
     private readonly Lazy<IReadOnlyCollection<IMethodInformation>> _interfaceDeclarations;
 
     private MethodInfoAdapter (MethodInfo methodInfo)
     {
       _methodInfo = methodInfo;
 
-      _cachedDeclaringType = new Lazy<ITypeInformation> (
+      _cachedDeclaringType = new Lazy<ITypeInformation?> (
```

Comment:
Then the initialization must be changed. The following is the current state:

```csharp
_declaringProperty = new Lazy<IPropertyInformation?> (
    () => PropertyInfoAdapter.CreateOrNull (_methodInfo.FindDeclaringProperty()),
    LazyThreadSafetyMode.ExecutionAndPublication);
```

As `.CreateOrNull(...)` is used, the return type of the `Func<>` must be nullable. This can be "fixed" by using `.Create(...)` and checking `_methodInfo.FindDeclaringProperty()` if you're sure that this would always find one.

Reply:
No, I meant that it's a TODO ticket to analyse if Create or CreateOrNull is the correct usecase here. I can't say either/or right now during the review.